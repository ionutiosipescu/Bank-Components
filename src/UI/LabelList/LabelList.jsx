import React from "react";
import { LabelListStyle } from "./LabelList.style";

function LabelList({ status, children }) {
  return <LabelListStyle status={status}>{children}</LabelListStyle>;
}

export default LabelList;
