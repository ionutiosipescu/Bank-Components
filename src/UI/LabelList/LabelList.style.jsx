import styled from "styled-components";

export const LabelListStyle = styled.label`
  font-family: "Heebo", sans-serif;
  font-size: 0.75rem;
  font-weight: 500;
  letter-spacing: 0.025rem;
  font-style: normal;
  border-radius: 1.25rem;
  padding: 0.35rem 0.75rem;
  box-shadow: 0px 1px 3px 1px rgba(51, 51, 51, 0.43);
  width: 100px;
  text-align: center;

  color: ${(props) =>
    props.status === "Completed"
      ? "var(--green)"
      : props.status === "Canceled"
      ? "var(--red)"
      : props.status === "Pending"
      ? "var(--gray-dark)"
      : "var(--gray-dark)"};
  background-color: ${(props) =>
    props.status === "Completed"
      ? "var(--green-light)"
      : props.status === "Canceled"
      ? "var(--red-light)"
      : props.status === "Pending"
      ? "var(--gray-light)"
      : "var(--gray-light)"};
`;
