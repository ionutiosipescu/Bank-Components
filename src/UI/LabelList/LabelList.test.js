import React from "react";
import { shallow } from "enzyme";
import LabelList from "./LabelList";

describe("LabelList component", () => {
  it("should render correctly with status prop", () => {
    const wrapper = shallow(
      <LabelList status="Completed">Test label</LabelList>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly without status prop", () => {
    const wrapper = shallow(<LabelList>Test label</LabelList>);
    expect(wrapper).toMatchSnapshot();
  });
});
