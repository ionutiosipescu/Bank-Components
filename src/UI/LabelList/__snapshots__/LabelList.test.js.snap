// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelList component should render correctly with status prop 1`] = `
<styled.label
  status="Completed"
>
  Test label
</styled.label>
`;

exports[`LabelList component should render correctly without status prop 1`] = `
<styled.label>
  Test label
</styled.label>
`;
